File: src/test/java/net/proxzi/StringsTest.java
package net.proxzi;

import org.junit.Test;

import static org.junit.Assert.*;

public class StringsTest {

	@Test
	public void testGenerateSlug() {
		assertEquals("hello", Strings.generateSlug("Hello"));
		assertEquals("hello-world", Strings.generateSlug("Hello-World"));
		assertEquals("hello-world", Strings.generateSlug("Hello World"));
		assertEquals("hello-world", Strings.generateSlug("Hello---World"));
		assertEquals("hello-world", Strings.generateSlug("Hello-_-World"));
		assertEquals("hello_world", Strings.generateSlug("Hello_--World"));
	}

}

File: src/test/java/net/proxzi/StringsTest.kt
package net.proxzi

import org.junit.Test
import kotlin.test.assertEquals

class StringsTest {

	@Test
	fun testGenerateSlug() {
		assertEquals("hello", "Hello".generateSlug())
		assertEquals("hello-world", "Hello-World".generateSlug())
		assertEquals("hello-world", "Hello World".generateSlug())
		assertEquals("hello-world", "Hello---World".generateSlug())
		assertEquals("hello-world", "Hello-_-World".generateSlug())
		assertEquals("hello_world", "Hello_--World".generateSlug())
	}

}
